logging/metadata/metadata.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package metadata provides access to metadata server adoported for needs
+// of the resource detection algorithm. The package partially duplicates
+// implementation of "cloud.google.com/go/compute/metadata" package.
+//
+// This package is a wrapper around the GCE metadata service,
+// as documented at https://developers.google.com/compute/docs/metadata.
+package metadata // import "cloud.google.com/go/logging/metadata"
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+)
+
+const (
+	// metadataIP is the documented metadata server IP address.
+	metadataIP = "169.254.169.254"
+
+	// metadataHostEnv is the environment variable specifying the
+	// GCE metadata hostname.  If empty, the default value of
+	// metadataIP ("169.254.169.254") is used instead.
+	// This variable name should be used when emulating the
+	// metadata server. Note that it is used by
+	// "cloud.google.com/go/compute/metadata" package.
+	metadataHostEnv = "GCE_METADATA_HOST"
+
+	userAgent = "gcloud-golang/0.1"
+
+	instancePathPrefix = "instance/attributes/"
+	projectPathPrefix  = "project/attributes/"
+)
+
+// NotDefinedError is returned when requested metadata is not defined.
+type NotDefinedError string
+
+func (suffix NotDefinedError) Error() string {
+	return fmt.Sprintf("metadata: attribute %q not defined", string(suffix))
+}
+
+// Error contains an error response from the server.
+type Error struct {
```

Comment:
Same comment as above.

Reply:
I will check if the use of this error is strictly internal and refactor the code if it is true.